docs/adr/0023-paypal-support.md

```diff
@@ -0,0 +1,104 @@
+# Add PayPal as a supported payment option
+
+- Deciders: Ben Bangert, Barry Chen, Tyler Duzan, Bianca Danforth, Se Yeon Kim, Wennie Leung
+- Date: 2020-11-11
+
+## Context and Problem Statement
+
+The Subscription Platform was implemented with Stripe as the payment processor and driver of subscription logic. Stripe is used as the source of truth for whether an account has subscriptions associated with it. We would like to add PayPal as a payment option for subscriptions, and integrate it appropriately with the existing system in a least-effort manner that is ideally flexible enough to support future expansions of payment processors, such as IAP.
+
+## Decision Drivers
+
+- Engineering resources
+- Future effort needed to integrate IAP
+- Code complexity as it relates to existing Subscription Platform infrastructure
+
+## Considered Options
+
+- Stripe-driven out-of-band invoice processing
+- PayPal subscriptions linked in FxA
+- Recurly
+
+## Decision Outcome
+
+Originally considered to be the 'fast-path', using PayPal subscriptions ended up being approximately the same engineering effort to implement as Stripe-driven invoice processing. Given additional user experience benefits of the Stripe-driven approach, it is now the recommend implementation path.
+
+## Pros and Cons of the Options
+
+### Stripe-driven out-of-band invoice processing
+
+Stripe defaults subscriptions to billing automatically using a payment source on file with Stripe. Stripe also provides the option to have subscriptions that have a `collection_method` of `send_invoice`. In this model, Stripe can track subscriptions, and generate invoices that can be paid `out-of-band`. This allows Stripe to have an accurate record of charges, invoices, and subscriptions but how the invoice is considered 'paid' is up to FxA.
+
+PayPal has a feature called reference transactions that allow FxA to get a customers authorization to charge their PayPal account. This authorization allows for later payments to be charged against the PayPal account. We will track this reference ID in Stripe metadata for the customer and mark the subscription as `send_invoice`.
```

Comment:
Excuse the quick drive-by, but I read through this ADR out of curiosity and noticed [PayPal's docs](https://developer.paypal.com/docs/archive/express-checkout/ec-set-up-reference-transactions/) on reference transactions state that the feature is deprecated as of January 2017. Am I missing something here?

Reply:
They are apparently not really deprecated, if you ask nicely. [Recurly uses this method to run PayPal transactions ](https://docs.recurly.com/docs/paypal-payments#section-get-approved-to-use-reference-transactions)and notes here: 

> Note: The message in PayPal’s documentation saying that Reference Transactions are deprecated is inaccurate. In fact, Reference Transactions are still fully supported for new and existing Recurly merchants who want to add PayPal.

Reply:
🤔 Well... that's interesting. If we're using this implementation in say, 3 years, that means we'll be using something marked as deprecated 7 years ago. Feels a bit funny, but perhaps they'll need to support it indefinitely.